File: Medium/2026-04-28-2033-Minimum-Operations-to-Make-a-Uni-Value-Grid/solution.js
function minOperations(grid, x) {
    const flatGrid = grid.flat();
    const minVal = Math.min(...flatGrid);
    const totalOps = flatGrid.reduce((acc, val) => {
        const diff = Math.abs(val - minVal);
        if (diff % x !== 0) return -1; // If not possible
        return acc + diff / x;
    }, 0);
    return totalOps;
}

File: Medium/2026-04-28-2033-Minimum-Operations-to-Make-a-Uni-Value-Grid/solution.py
class Solution:
    def minOperations(self, grid: List[List[int]], x: int) -> int:
        flat_grid = [num for row in grid for num in row]
        min_val = min(flat_grid)
        total_operations = 0
        
        for num in flat_grid:
            if (num - min_val) % x != 0:
                return -1
            total_operations += (num - min_val) // x
        
        return total_operations